### PR TITLE
strip extraneous leading 0000 from pci_bus_id

### PR DESCRIPTION
**Pull request by altair4**
#### Bug/feature Description
To quote the discoverer of the problem, "The device id=... reported by nvidia-smi -q -x latest incarnation now returns a PCI domain written with 8 hex characters (despite the fact the PCI domain is 16 bit)."

#### Affected Platform(s)
Certain Nvidia GPUs.

#### Cause / Analysis / Design
Mislabeled data reported by certain Nvidia GPU devices.

#### Solution Description
After parsing the XML returned by nvidia-smi,  make sure the PCI domain is 16 bits and 4 hex numbers.
To verify the solution, one must first locate a problematic GPU to test.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [ ] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by mike0042** [comment]
Code looks good @altair4. A few things:

- Please do the needful with regard to CLA.
- Please correct the title of the PR. (Use the PP ticket number.)
- Please run the PTL tests and attach the log either to the PR or the ticket.

File: src/hooks/cgroups/pbs_cgroups.PY [hunk 1]
```diff
@@ -1282,6 +1282,10 @@ class NodeConfig(object):
             for child in root:
                 if child.tag == "gpu":
                     bus_id = child.get('id')
+                    # make sure the PCI domain is 16 bits and 4 hex numbers
+                    pci_domain, pci_rest = child.get('id').split(":", 1)
+                    fixed_pci_domain = pci_domain[-4:]
+                    bus_id = fixed_pci_domain + ":" + pci_rest
```
Comment (arungrover) [inline src/hooks/cgroups/pbs_cgroups.PY]: @altair4 since bus_id is being set here then line 1284 is not needed right?

Comment (altair4) [reply src/hooks/cgroups/pbs_cgroups.PY]: Right - it's now been removed.

Comment (mike0042) [reply src/hooks/cgroups/pbs_cgroups.PY]: Please push your commit so we can see the change.

**Comment by altair4** [comment]
Withdrawn since it will be subsumed by a larger pull request from Mike Karo.

**Comment by altair4** [comment]
Withdrawn since it will be subsumed by a larger pull request from Mike Karo.

**Pull request closed without merging**